docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+
+```{note}
+it's recommended that you avoid using jQuery as this will be removed in a future version of Wagtail.
+```
+
+(extending_client_side_injecting_javascript)=
+
+## Adding custom JavaScript
+
+Within Wagtail's admin interface, there are a few ways to add JavaScript.
+
+The simplest way is to add global JavaScript files via hooks, see [](insert_editor_js) and [](insert_global_admin_js).
+
+For JavaScript added when a specific Widget is used you can add an inner `Media` class to ensure that the file is loaded when the widget is used, see [Django's docs on their form `Media` class](https://docs.djangoproject.com/en/stable/topics/forms/media/#assets-as-a-static-definition).
+
+In a similar way, Wagtail's [template components](template_components) provide a `media` property or `Media` class to add scripts when rendered.
+
+These will ensure the added files are used in the admin after the core JavaScript admin files are already loaded.
+
+(extending_client_side_using_events)=
+
+## Extending with DOM events
+
+When approaching client-side customisations or adopting new components, try to keep the implementation simple first, you may not need any knowledge of Stimulus, React, JavaScript Modules or a build system to achieve your goals.
+
+The simplest way to attach behaviour to the browser is via DOM Events and plain (vanilla) JavaScript.
+
+For example, if you want to attach some logic to a field value change in Wagtail you can add an event listener, check if it's the correct element and change what you need.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('change', function (event) {
+    if (event.target) {
+        console.log('field has changed', event.target);
+    }
+});
+```
+
+Alternatively, you can write JavaScript logic to achieve this by listening to all click events and checking if the clicked element corresponds to the comments section.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('click', function (event) {
+    if (!event.target) return;
+
+    isCommentsToggle = event.target.matches(
+        '[data-side-panel-toggle="comments"]',
+    );
+
+    if (!isCommentsToggle) return;
+    console.log('Comments have been toggled');
+});
+```
+
+Then, you can register this with the `insert_global_admin_js` hook.
+
+```python
+# myapp/wagtail_hooks.py
+from django.templatetags.static import static
+from django.utils.safestring import mark_safe
+
+from wagtail import hooks
+
+@hooks.register('insert_global_admin_js')
+def global_admin_js():
+    return mark_safe(
+        f'<script src="{static("js/events.js")}"></script>',
+    )
+```
+
+### Wagtail's custom DOM events
+
+-   Wagtail supports some custom behaviour via listening or dispatching custom DOM events, usually with the prefix `wagtail:` or `w-` for specific Stimulus controllers.
```

Comment:
Here I think my preference would be to document the specific events we want people to reuse, rather than setting a pattern of allowing reliance on any and all DOM events with those prefixes. It feels like too much busywork for us to treat all prefixed DOM events as needing to remain API-compatible. If an event is important enough for third-party customisations, then I think having it explicitly named and described in the docs is a good baseline.

Reply:
Great, I'll remove this.